src/DwitTech.AccountService.Core/Services/UserService.cs

```diff
@@ -1,10 +1,101 @@
-﻿using DwitTech.AccountService.Core.Interfaces;
+﻿using AutoMapper;
+using DwitTech.AccountService.Core.Dtos;
+using DwitTech.AccountService.Core.Interfaces;
+using DwitTech.AccountService.Core.Utilities;
 using DwitTech.AccountService.Data.Entities;
 using DwitTech.AccountService.Data.Repository;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
 
 namespace DwitTech.AccountService.Core.Services
 {
     public class UserService : IUserService
     {
+
+        private readonly IUserRepository _userRepository;
+        private readonly IRoleRepository _roleRepository;
+        private readonly ILogger<UserService> _logger;
+        private readonly IActivationService _activationService;
+        private readonly IEmailService _emailService;
+        public UserService(IUserRepository userRepository, IRoleRepository roleRepository, ILogger<UserService> logger, IActivationService activationService, IEmailService emailService)
+        {
+            _userRepository = userRepository;
+            _roleRepository = roleRepository;
+            _logger = logger;
+            _activationService = activationService;
+            _emailService = emailService;
+        }
+
+        private async Task<Data.Entities.Role> GetAssignedRole(UserDto user)
+        {
+            var roles = await _roleRepository.GetRoles();
+            if (user.Roles.ToString() is not null)
+            {
+                var assignedRole = roles.FirstOrDefault(x => x.Id == ((int)user.Roles));
+                if (assignedRole != null)
+                {
+                    return assignedRole;
+                }
+            }
+            return roles.First(x => x.Id == 2);
+
+        }
+
+        private User CustomMapper(UserDto user, Data.Entities.Role userIdentifiedRole)
```

Comment:
If I remember correctly, boss said methods should be named like verbs. CustomMapper is named like a noun. It should follow the convention used in your previous methods, e.g GetCustomMap